pyodide-build/pyodide_build/buildall.py

```diff
@@ -371,7 +371,7 @@ def build_from_graph(pkg_map: dict[str, BasePackage], outputdir: Path, args) ->
     built_queue: Queue = Queue()
     thread_lock = Lock()
     queue_idx = 1
-    package_set = {}
+    package_set: dict[str, None] = {}  # TODO: shoudn't this be a set?
```

Comment:
Shouldn't this be a set? IT's just assigning Nones, and the name literally includes "set".

Reply:
I think iteration over dictionaries is in insertion order whereas iteration over sets is not. I believe this is why we use a dictionary here.